src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -25,7 +24,6 @@ public class SocketInput : ICriticalNotifyCompletion
         private MemoryPoolBlock2 _head;
         private MemoryPoolBlock2 _tail;
         private MemoryPoolBlock2 _pinned;
-        private readonly object _sync = new Object();
```

Comment:
Love it when locks can be removed :+1: Are you 100% positive this won't introduce any race conditions?